ecosystem/purescript-0.14-migration-guide.md

```diff
@@ -0,0 +1,622 @@
+# PureScript 0.14 Migration Guide
+
+This guide summarizes the changes you may need to make to your code to migrate from PureScript 0.13 to PureScript 0.14. It covers major changes to the compiler, the core libraries, and tooling.
+
+Related resources:
+- The PureScript 0.14 [release announcement]()
+- The PureScript 0.14 [compiler release notes]()
+
+Compiler releases are often accompanied by breaking changes in the core libraries. While some major library changes are described in this document, you should consult the individual changelogs for any libraries you depend on.
+
+**Contents**
+
+1. [Compiler Changes](#compiler-changes)
+1. [Library Changes](#library-changes)
+1. [Tooling Changes](#tooling-changes)
+
+## Compiler Changes
+
+The PureScript 0.14 compiler introduces a number of new features, including support for polymorphic kinds and zero-cost coercions. These features may require changes to your code.
+
+This section of the guide will walk through common changes you may need to make, but it isn't a complete overview of changes in the 0.14 compiler and doesn't provide an explanation of these new features. You should also consult the [PureScript 0.14 compiler release notes]().
+
+### Polykinds & Type In Type
+
+PureScript now supports polymorphic kinds and kind signatures. Most code is unaffected, but there are some changes.
+
+The type checker now supports `TypeInType`, so the old `Kind` data type and namespace has been removed. The `foreign import kind` syntax is deprecated, so this will warn:
+
+```purs
+foreign import kind Boolean
+foreign import data True :: Boolean
+foreign import data False :: Boolean
+```
+
+and should be rewritten as an empty data declaration:
+
+```purs
+data Boolean
+foreign import data True :: Boolean
+foreign import data False :: Boolean
+```
+
+The old unary `# Type` syntax for row kinds is deprecated, and it should instead be written as `Row Type`.
+
+```diff
+- class ListToRow (list :: RowList) (row :: # Type)
++ class ListToRow (list :: RowList) (row :: Row Type)
```

Comment:
I think it would be good to explicitly say that the kind of `RowList` has changed from `Type` to `Type -> Type`, where the argument is the kind of the labels in the row list, so if you had  `RowList` in 0.13.x code, that should now be written as `RowList Type`.